include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_radix_sort_one_wg.h

```diff
@@ -173,10 +173,23 @@ struct __subgroup_radix_sort
                         uint16_t __begin_bit = 0;
                         constexpr uint16_t __end_bit = sizeof(_KeyT) * ::std::numeric_limits<unsigned char>::digits;
 
+                        auto __val_mem_adjusted_barrier = [__it]() {
+                            if constexpr (_SLM_tag_val::value)
+                                __dpl_sycl::__group_barrier(__it);
+                            else
+                                __dpl_sycl::__group_barrier(__it, __dpl_sycl::__fence_space_global_and_local{});
+                        };
+                        auto __count_mem_adjusted_barrier = [__it]() {
+                            if constexpr (_SLM_counter::value)
+                                __dpl_sycl::__group_barrier(__it);
+                            else
+                                __dpl_sycl::__group_barrier(__it, __dpl_sycl::__fence_space_global_and_local{});
```

Comment:
Since `__counters` is entirely through global memory in this case, is `__dpl_sycl::__fence_space_global` sufficient here?

Reply:
Indeed, `__fence_space_global` is sufficient. I've applied your suggestion.